### PR TITLE
feat: added plugin to search for xdg user directories

### PR DESCRIPTION
**Pull request by abenz1267**
Hi,

this plugin would allow you to use anyrun to search/open xdg user directories, see [here](https://wiki.archlinux.org/title/XDG_user_directories).

What do you think? Personally i use the finder rather often in order to open those directories.

Regards

### CODE REVIEW

**Comment by Kirottu** [comment]
It feels a bit redundant as the Kidex plugin already achieves file/folder search.

Also, is that `~/.config/user-dirs.dirs` a standard file?

**Comment by abenz1267** [comment]
> It feels a bit redundant as the Kidex plugin already achieves file/folder search.

I actually tried the Kidex plugin before i created this and my reasons where:
- i don't need filesearch
- i don't want to have a background process running for something as simple as looking up user directories
- i wanted something really tiny in scope

> Also, is that `~/.config/user-dirs.dirs` a standard file?

Yes, it's the default file for xdg-user-dir.

If you feel like this is redundant and you don't want it in here, that's no problem. I just thought other people might be interested as well.

Regards

**Comment by Kirottu** [comment]
Having an entire plugin just for this purpose seems a bit redundant, a general shortcut plugin would probably be more useful in my opinion.

**Comment by abenz1267** [comment]
> Having an entire plugin just for this purpose seems a bit redundant, a general shortcut plugin would probably be more useful in my opinion.

I thought about that as well, but then you'd either: 
a) need to implement logic to find shortcuts/bookmarks for different "platforms", e.g. nautilus saves bookmarks in a different place then dolphin
b) need to define the shortcuts in a config file for the plugin

But now if you use xdg-user-dirs anyways and define commonly used folders there as well already, you are kinda doing double the work. I'm not sure.

My original intent was indeed to just search for nautilus bookmarks. But going the xdg-user-dirs approach looked saner in my eyes, as it's independent from any installed filebrowser gui.

Regards

**Comment by Kirottu** [comment]
I'd lean towards configuring shortcuts inside a config file, then `xdg-open` could be used to open them so you could add shortcuts to basically anything you wish.

File: plugins/userdirs/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    dirs: Vec<String>,
+}
+
+#[init]
+fn init(_config_dir: RString) -> State {
+    let mut dirs: Vec<String> = vec![];
+
+    match home::home_dir() {
+        Some(path) => {
+            if let Ok(lines) = read_lines(format!("{}/.config/user-dirs.dirs", path.display())) {
```
Comment (Sntx626) [inline plugins/userdirs/src/lib.rs:22]: Not sure if this is present on every system. It is present on mine (auto-generated by nix).

**Review by Sntx626 — CHANGES_REQUESTED** [review]
I'd like to see a backup in case `~/.config/user-dirs.dirs` is not present.

The hard-to-read part of the code (look at my other comments) should be simplified.

Overall this plugin might be redundant with kidex, but yet again not everyone is running kidex. So it's still a viable plugin.

**Pull request closed without merging**